tools/sigclient/pkg/alerts/alerts_loadtest.go

```diff
@@ -0,0 +1,321 @@
+		MinuteData: make(map[int]MinuteSummaryData),
+	}
+
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	timeoutDuration := 2 * time.Minute
+	timeout := time.NewTimer(timeoutDuration)
+	defer timeout.Stop()
+
+	for {
+		currentMinute := int(time.Since(startTime).Minutes())
+		select {
+		case webhookBody := <-webhookChan:
+			alertReceivedTime := time.Now()
+			timeout.Stop() // Stop the timeout as we received an alert
+			log.Infof("CurrentMinute=%v,ReceivedAlert=%v,NumEvaluations=%v", currentMinute, webhookBody.Title, webhookBody.NumEvaluationsCount)
+
+			alertEvalMinute := int(webhookBody.NumEvaluationsCount) - 1
+
+			minuteData, exists := summary.MinuteData[alertEvalMinute]
+			if !exists {
+				minuteData = MinuteSummaryData{
+					Alerts: make(map[string]AlertData),
+				}
+			}
+
+			delay := alertReceivedTime.Sub(startTime.Add(time.Duration(alertEvalMinute) * time.Minute))
+
+			alertData := minuteData.Alerts[webhookBody.Title]
+			alertData.ReceivedCount++
+			alertData.TotalDelay += delay
+			minuteData.Alerts[webhookBody.Title] = alertData
+			minuteData.TotalAlertsReceived++
+			minuteData.TotalDelay += delay
+
+			summary.TotalAlertsReceived++
+			summary.TotalDelay += delay
+
+			summary.MinuteData[alertEvalMinute] = minuteData
+
+			timeout.Reset(timeoutDuration) // Reset the timeout so that we won't trigger timeout and exit.
+		case <-ticker.C:
+			// Log summary of the previous minute
+			minute := currentMinute - 1
+			if minute >= 0 {
+				if minuteData, exists := summary.MinuteData[minute]; exists {
+					minuteData.AverageDelay = minuteData.TotalDelay / time.Duration(minuteData.TotalAlertsReceived)
+					logMinuteSummary(minute, minuteData, numAlerts)
+					summary.MinuteData[minute] = minuteData
+				}
+			}
+		case <-timeout.C:
+			log.Errorf("Timed out waiting for alerts")
+			doneChan <- false
+			return
+		case <-exitChan:
+			log.Warnf("Received Exit Signal. Exiting the test!")
+
+			if summary.TotalAlertsReceived > 0 {
+				summary.AverageDelay = summary.TotalDelay / time.Duration(summary.TotalAlertsReceived)
+				summary.AverageAlertsPerMinute = float64(summary.TotalAlertsReceived) / float64(currentMinute+1)
+			} else {
+				summary.AverageDelay = 0
+				summary.AverageAlertsPerMinute = 0
+			}
+
+			log.Infof("Final Summary: Total Time=%v", time.Since(startTime))
+			logSummary(summary)
+			doneChan <- true
+			return
+		}
+	}
+}
+
+func logMinuteSummary(minute int, minuteData MinuteSummaryData, numAlerts int) {
+	alertsCount := len(minuteData.Alerts)
+	log.Infof("Minute=%d,IsSummary=%v,UniqueAlertsCount=%d,Pass=%v,TotalAlertsCount=%v,AverageDelay=%v", minute, true, alertsCount, alertsCount >= numAlerts, minuteData.TotalAlertsReceived, minuteData.AverageDelay)
+	for title, data := range minuteData.Alerts {
+		averageDelay := data.TotalDelay / time.Duration(data.ReceivedCount)
+		log.Infof("Minute=%d,Alert=%s,ReceivedCount=%d,AverageDelay=%v", minute, title, data.ReceivedCount, averageDelay)
```

Comment:
nit: add a space after the commas in the strings, just so it's a little easier to read for someone looking at the output logs

Also for `AverageDelay` it would be good to indicate the units of time (or maybe that already gets printed?)

Reply:
For `AverageDelay`, the unit is already printed. I did not added space specifically because, I am going to tunnel these logs to SigLens and parse them as `csv` through vector and then use `parse_key_value` so that they will be properly formatted into columns.